### PR TITLE
Compatible with webhook.v2 of coding painlessly

### PR DESCRIPTION
**Pull request by wusisu**

### CODE REVIEW

File: LICENSE_HEADER [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Jenkins plugin for Coding https://coding.net
 
-Copyright (C) 2016-2018 Shuanglei Tao <tsl0922@gmail.com>
+Copyright (c) 2016-2018 Shuanglei Tao <tsl0922@gmail.com>
+Copyright (c) 2016-present, Coding, Inc.
```
Comment (wyw) [inline LICENSE_HEADER:4]: `CODING Tech`?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Sat Jun 25 15:17:58 CST 2016
+#Thu Feb 15 20:12:54 CST 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.14-all.zip
```
Comment (wyw) [inline gradle/wrapper/gradle-wrapper.properties]: Package `bin` is enough.

File: src/main/java/net/coding/jenkins/plugin/bean/WebHookTask.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public CodingBaseEvent getBaseEvent() {
+        switch (event) {
+            case EVENT_PUSH:
+                return push;
+            case EVENT_MERGE_REQUEST:
+                return mergeRequest;
+            default:
+                return null;
```
Comment (wyw) [inline src/main/java/net/coding/jenkins/plugin/bean/WebHookTask.java]: How about `return new CodingBaseEvent();`?

File: src/main/java/net/coding/jenkins/plugin/bean/WebHookTask.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public User getSender() {
+        return getBaseEvent().getSender();
+    }
+
+    public CodingBaseEvent getBaseEvent() {
+        switch (event) {
+            case EVENT_PUSH:
+                return push;
```
Comment (wyw) [inline src/main/java/net/coding/jenkins/plugin/bean/WebHookTask.java:58]: The `push` object may be null?

File: src/main/java/net/coding/jenkins/plugin/listener/MergeRequestRunListener.java [hunk 5]
```diff
@@ -132,12 +147,27 @@ private void addResultNote(String apiToken, String buildUrl, String commitId, bo
...
+            // is professional
+            return String.format("%s/api/user/%s/project/%s", matcher.group(1), matcher.group(2), matcher.group(3));
+        }
+        pattern = Pattern.compile("(https?://[^/]+)/p/([^/]+).*");
+        matcher = pattern.matcher(htmlUrl);
+        if (matcher.matches()) {
+            // is enterprise
+            String host = matcher.group(1);
+            String projectName = matcher.group(2);
+            System.out.println(host);
```
Comment (wyw) [inline src/main/java/net/coding/jenkins/plugin/listener/MergeRequestRunListener.java]: Aha.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (wusisu) [reply gradle/wrapper/gradle-wrapper.properties]: how to change this ...

Comment (wyw) [reply gradle/wrapper/gradle-wrapper.properties]: Keep the original one.

File: LICENSE_HEADER (hunk 1 above)
Comment (wusisu) [reply LICENSE_HEADER:4]: Copy from https://github.com/facebook/react/blob/master/LICENSE

What is our company's name?

File: src/main/java/net/coding/jenkins/plugin/CodingPushTrigger.java [hunk 6]
```diff
@@ -113,11 +115,16 @@ protected Object readResolve() throws ObjectStreamException {
         return super.readResolve();
     }
 
-    public void onPost(WebHook webHook, String event) {
-        if (StringUtils.isEmpty(webHookToken) || StringUtils.equals(webHookToken, webHook.getToken())) {
-            triggerHandler.handle(job, webHook, event, ciSkip);
+    public void onPost(WebHookTask task) {
+        IWebHookHelper helper = CodingWebHook.webHookHelper(task.getVersion());
+        if (helper == null) {
+            return;
```
Comment (tsl0922) [inline src/main/java/net/coding/jenkins/plugin/CodingPushTrigger.java]: `throw hudson.util.HttpResponses.error(400, "Bad Request");`

File: src/main/java/net/coding/jenkins/plugin/listener/MergeRequestRunListener.java [hunk 7]
```diff
@@ -132,12 +147,27 @@ private void addResultNote(String apiToken, String buildUrl, String commitId, bo
...
+            Matcher enterprise = Pattern.compile("https?://([^.]+).*").matcher(host);
+            if (enterprise.matches()) {
+                return String.format("%s/api/user/%s/project/%s", host, enterprise.group(1), projectName);
+            }
         }
-        return String.format("https://coding.net/api/user/%s/project/%s", parts[0], parts[1]);
+        return "";
```
Comment (tsl0922) [inline src/main/java/net/coding/jenkins/plugin/listener/MergeRequestRunListener.java]: `throw new IllegalArgumentException("Invalid project api url: " + htmlUrl);`

File: src/main/java/net/coding/jenkins/plugin/bean/WebHookTask.java (hunk 4 above)
Comment (wusisu) [reply src/main/java/net/coding/jenkins/plugin/bean/WebHookTask.java:58]: Should not null when it comes here.

**Pull request merged**